### DISCUSSION TITLE
포스트 프로세싱 관련 이슈 질문

### DISCUSSION THREAD

**Discussion by ted002** [naverz/zepeto-studio-kor#882] [5. Q&A]
포스트 프로세싱의 프로파일 관련 코드들중 GetSetting과 TryGetSettings의 경우 빌드시에 코드들이 제대로 확인되지 않는 경우가 보이는데, 이건 오류인가요, 아니면 제가 제대로 사용하지 못하는건가요?

타입 스크립트에 작성한 코드
![1](https://user-images.githubusercontent.com/73985764/173516872-4d6d5a5a-579d-4add-b917-b9c8c688096c.PNG)

Open Compiled Script로 열어본 코드의 상태
![2](https://user-images.githubusercontent.com/73985764/173517024-9160f162-1305-497e-baa5-22fc8cb56e6e.PNG)

이 두번째 사진처럼, 컴파일드 스크립트에는 HasSettings에서는 제가 원하는 AmbientOcclusion세팅을 가져올수 있는 스크립트가 컴파일드 스크립트에 있는데, GetSetting과 TryGetSettings에는 타입스크립트에서 작성한 템플릿이 사라져 있고 HasSettings처럼 구별하기 위한 번역도 없더군요.

물론 유니티에서도 오류가 납니다.
유니티 오류 전문
![3](https://user-images.githubusercontent.com/73985764/173517779-2911528e-1241-4904-af21-50e92c7247d6.PNG)

해서, index.d.ts측을 확인해보니 HasSettings는 

        public HasSettings<T extends PostProcessEffectSettings>():boolean;
        
        public HasSettings($type: System.Type):boolean;

두가지의 오버라이드가 있는데, GetSetting과 TryGetSettings는 <T extends PostProcessEffectSettings>하나씩뿐이더군요.

![4](https://user-images.githubusercontent.com/73985764/173518025-68c14dd2-8f0d-4e94-8089-340b74258935.PNG)

기능 미지원인지, 버그인지, 아니면 제 사용 미스인지 확인부탁드립니다

**Comment by Deytooh** [comment]
PostProcessVolume.sharedProfile 로 하면 어때요 ? 전 .profile 안돼서 sharedProfile 썼는데 됐어요

**Reply by ted002** [reply]
오 한번 써보겠습니다

**Reply by ted002** [reply]
PostProcessVolume.sharedProfile.sharedProfile을 써도 여전히 GetSetting은 안되네요. 

혹시 괜찮으시다면 어떤식으로 쓰셨는지 알 수 있을까요? 

저는 이거 Public으로 멤버변수 볼륨을 가져오고 GetSetting을 하는데 GetSetting<Bloom>()으로 해도 뭐 다른  이펙트로 해도 다 동일하게  GetSetting is not a function이 뜨더군요. 

참고로 제페토 유니티 버전은 1.4.1 이고, 스크립트는 1.3.2버전입니다. 

GetSetting이 있는 index.d.ts 내의 클래스는 PostProcessProfile클래스입니다.
